server/src/main/java/org/opensearch/cluster/routing/IndexShardRoutingTable.java

```diff
@@ -738,7 +738,7 @@ public boolean equals(Object o) {
         IndexShardRoutingTable that = (IndexShardRoutingTable) o;
 
         if (!shardId.equals(that.shardId)) return false;
-        if (!shards.equals(that.shards)) return false;
+        if (!new HashSet<>(shards).equals(new HashSet<>(that.shards))) return false;
```

Comment:
Would this work
```
first.size() == second.size() && first.containsAll(second) && second.containsAll(first)
```

Reply:
This would work, but using HashSet seems to be more optimized with these checks.

Reply:
Changed to suggested approach to avoid memory overhead